modules/services/cloud-bench/main.tf

```diff
@@ -1,110 +1,19 @@
-###################################################
-# Fetch & compute required data
-###################################################
-
-data "sysdig_secure_trusted_cloud_identity" "trusted_identity" {
-  cloud_provider = "gcp"
-}
-
-data "google_project" "project" {
-  project_id = var.project_id
-}
-
 locals {
-  regions_scope_clause = length(var.regions) == 0 ? "" : " and gcp.region in (\"${join("\", \"", var.regions)}\")"
-}
-
-###################################################
-# Configure Sysdig Backend
-###################################################
-
-resource "sysdig_secure_cloud_account" "cloud_account" {
-  account_id     = data.google_project.project.number
-  alias          = data.google_project.project.name
-  cloud_provider = "gcp"
-  role_enabled   = "true"
-  role_name      = var.role_name
-}
-
-resource "sysdig_secure_benchmark_task" "benchmark_task" {
-  name     = "Sysdig Secure for Cloud (GCP) - ${data.google_project.project.name}"
-  schedule = "0 6 * * *"
-  schema   = "gcp_foundations_bench-1.2.0"
-  scope    = "gcp.projectId = \"${data.google_project.project.number}\"${local.regions_scope_clause}"
-
-  # Creation of a task requires that the Cloud Account already exists in the backend, and has `role_enabled = true`
-  depends_on = [sysdig_secure_cloud_account.cloud_account]
-}
-
-###################################################
-# Create Service Account and setup permissions
-###################################################
-
-resource "google_service_account" "sa" {
-  project = var.project_id
-
-  account_id   = var.role_name
-  display_name = "Service account for cloud-bench"
+  project_ids = var.project_id == "" ? var.project_ids : [var.project_id]
 }
 
-resource "google_project_iam_member" "viewer" {
-  project = var.project_id
+module "trust_relationship" {
+  for_each = toset(var.project_ids)
+  source   = "./trust_relationship"
 
-  role   = "roles/viewer"
-  member = "serviceAccount:${google_service_account.sa.email}"
+  project_id = each.key
```

Comment:
Lets keep role_name as a variable for both the `cloud_bench` module _and_ the `trust_itentity` module, and pass it down here. That way the role_name can be specified in the examples

Reply:
Fixed!

Reply:
We'll need to add an explicit `role_name = var.role_name` to each of these submodules to propagate it down I'm pretty sure